packages/devtools/lib/src/main.dart

```diff
@@ -148,27 +151,41 @@ class PerfToolFramework extends Framework {
     }
   }
 
-  void _initHotReloadServiceListener() {
-    final hotReloadServiceName = registrations.hotReload.service;
-    serviceManager.hasRegisteredService(hotReloadServiceName,
-        (bool reloadServiceAvailable) {
-      if (reloadServiceAvailable) {
-        _buildReloadRestartButtons();
-      } else {
-        clearGlobalActions();
-      }
-    });
+  void _initHotReloadRestartServiceListeners() {
+    serviceManager.hasRegisteredService(
+      registrations.hotReload.service,
+      (bool reloadServiceAvailable) {
+        if (reloadServiceAvailable) {
+          _buildReloadButton();
+        } else {
+          removeGlobalAction(_reloadTooltip);
+        }
+      },
+    );
+
+    serviceManager.hasRegisteredService(
+      registrations.hotRestart.service,
+      (bool reloadServiceAvailable) {
+        if (reloadServiceAvailable) {
+          _buildRestartButton();
+        } else {
+          removeGlobalAction(_restartTooltip);
+        }
+      },
+    );
   }
 
-  void _buildReloadRestartButtons() async {
+  void _buildReloadButton() async {
     // TODO(devoncarew): We currently create hot reload events when hot reload
     // is initialed, and react to those events in the UI. Going forward, we'll
     // want to instead have flutter_tools fire hot reload events, and react to
     // them in the UI. That will mean that our UI will update appropriately
     // even when other clients (the CLI, and IDE) initial the hot reload.
 
-    final ActionButton reloadAction =
-        ActionButton('icons/hot-reload-white@2x.png', 'Hot Reload');
+    final ActionButton reloadAction = ActionButton(
+      'icons/hot-reload-white@2x.png',
+      _reloadTooltip,
```

Comment:
Nit: I would move the tip text back to the constructor call. Having tip text right next to where it is used makes the code easier to read.

Reply:
I made it a variable so that we can use it in `_initHotReloadRestartServiceListeners`. I don't think we should maintain two different strings that need to be consistent. `removeGlobalAction` uses the tooltip text to match the correct button to remove.

Reply:
totally fine if there are multiple uses. I'd missed that while reviewing.